### ISSUE TITLE
Set `Requeued` condition, and unset `QuotaReserved` condition on eviction in Workload Reconciler

### ISSUE THREAD

**Issue by PBundyra** [kubernetes-sigs/kueue#3474]
**What would you like to be cleaned**:
Currently the `Requeued`, and `QuotaReserved` conditions are set by the [jobframework reconciler](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/reconciler.go#L451-L471)

Ideally this logic should be moved to the Workload Controller to simplify logic, ease maintenance and adding new features, and avoid unpredictable race conditions. The logic should be atomic with eviction

**Why is this needed**:

**Comment by mimowo** [comment]
I agree it would be cleaner, but for that we need to figure out how solve the dependency issue on the Job, as currently workload controller is Job-agnostic: https://github.com/kubernetes-sigs/kueue/blob/e5dd891432f61c4a418e69405c840110553c07b0/pkg/controller/jobframework/reconciler.go#L458.

**Comment by tenzen-y** [comment]
As far as I know, during introducing this mechanism, we evaluated this proposed solution. But, we declined that based on the  Michal mentioned reason. I guess that we need to introduce another mechanism or a lot of refactoring...

**Comment by Horiodino** [comment]
If this is open, i would love to take it forward.

**Comment by Horiodino** [comment]
If this is open, i would like to take it forward.

**Comment by PBundyra** [comment]
Hi @Horiodino, great! Thanks for expressing interest. As this issue requires a lot of changes, please let us know in advance how to plan to tackle this

**Comment by Horiodino** [comment]
Is there any diagram or documentation for the JobFramework and Workload? Also, can you point me to where the Workload Reconciler is implemented?

**Comment by PBundyra** [comment]
Unfortunately we don't have any diagram. Here's the Workload's reconciler implementation: https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/core/workload_controller.go#L137

**Comment by Horiodino** [comment]
1. Use Workload Conditions/Spec Fields: The `WorkloadReconciler` uses `workload.IsActive(&wl)` to determine job activity by checking `wl.Spec.Active` and the absence of the `WorkloadFinished` condition. The `JobFramework Reconciler` updates `wl.Spec.Active` to `false` when stopping a job (e.g., in `stopJob`) or setting `WorkloadFinished`. On eviction, the `WorkloadReconciler` sets `WorkloadEvicted` and may reset admission, implicitly assuming `wl.Spec.Active` reflects job state. Replace direct calls to `job.IsActive()` with checks against `wl.Spec.Active` and conditions like `WorkloadFinished`. Ensure `JobFramework Reconciler` reliably syncs job state to `wl.Spec.Active`, reducing dependency on direct job queries.

2. Introduce Job State Field in Workload Status: Add a new field, `Status.JobActive`, to `kueue.Workload`. The `JobFramework Reconciler` sets `wl.Status.JobActive = job.IsActive()` during reconciliation. The `WorkloadReconciler` checks `wl.Status.JobActive` rather than calling `job.IsActive()`. Decouples job state checks from job-specific logic, centralizing status management within `Workload`. Enhances readability and traceability of job state. Update the `kueue.Workload` spec to include `Status.JobActive`. Ensure the `JobFramework Reconciler` sets this field during state transitions. Adjust `workload.IsActive()` to use `Status.JobActive`.

3. Event-Driven Job State Notification: Leverage event-based communication between `JobFramework Reconciler` and `WorkloadReconciler`. The `WorkloadReconciler` already has support for notifications through `WorkloadUpdateWatcher` (via `NotifyWorkloadUpdate`). Add a watcher mechanism in `WorkloadReconciler` to receive job state change events. The `JobFramework Reconciler` triggers `NotifyWorkloadUpdate` when job state changes (e.g., active to inactive). Introduce a new event type for job state changes. Extend `WorkloadReconciler` to handle job state events and update workload conditions accordingly.

Wdyt folks?
